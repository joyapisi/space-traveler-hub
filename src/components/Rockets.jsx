import { useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { getRockets, reserveRocket } from '../redux/rockets/rocketSlice';

const Rockets = () => {
  const dispatch = useDispatch();
  const { rocket } = useSelector((store) => store.rocket);
  const { isLoading } = useSelector((store) => store.rocket);

  useEffect(() => {
    if (isLoading === false) dispatch(getRockets());
  }, [dispatch, isLoading]);

  return (
    <div className="rockets-container">
      {isLoading && rocket.map((rocket) => (
        <div key={rocket.id} className="rocket-card">
          <img src={rocket.image} alt={rocket.name} className="rocket-image" />
          <h2 className="rocket-title">{rocket.name}</h2>
          <p className="rocket-details">
            {rocket.reserved && <span className="reserved">Reserved</span>}
            {rocket.description}
          </p>
          <button
            type="button"
            className={rocket.reserved ? 'unreserve' : 'reserve'}
            onClick={() => dispatch(reserveRocket(rocket.id))}
          >
            {rocket.reserved ? 'cancel reserve' : 'reserve rocket'}
          </button>
        </div>
      ))}
    </div>
  );
};

export default Rockets;
