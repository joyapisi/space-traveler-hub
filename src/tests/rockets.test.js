import { Provider } from 'react-redux';
import TestRenderer from 'react-test-renderer';
import Rockets from '../components/Rockets';
import Store from '../redux/store';
import '@testing-library/jest-dom/extend-expect';

describe('Rockets', () => {
  test('Render Rockets Component', () => {
    const tree = TestRenderer.create(
      <Provider store={Store}>
        <Rockets />
      </Provider>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
