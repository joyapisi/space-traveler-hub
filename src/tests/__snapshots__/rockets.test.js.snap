// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rockets Render Rockets Component 1`] = `
<div
  className="flex rockets-container"
/>
`;
